spec/invoice_item_spec.rb

```diff
@@ -0,0 +1,145 @@
+require 'rspec'
+require './lib/invoice_item'
+
+describe InvoiceItem do
+  describe '#initialize' do
+    it 'exists' do
+      ii_details = {
+        id: 6,
+        item_id: 7,
+        invoice_id: 8,
+        quantity: 1,
+        unit_price: BigDecimal(10.99, 4),
+        created_at: Time.now,
+        updated_at: Time.now
+      }
+      invoice_item = InvoiceItem.new(ii_details)
+
+      expect(invoice_item).is_a? InvoiceItem
+    end
+
+    it 'returns the integer id' do
+      ii_details = {
+        id: 6,
+        item_id: 7,
+        invoice_id: 8,
+        quantity: 1,
+        unit_price: BigDecimal(10.99, 4),
+        created_at: Time.now,
+        updated_at: Time.now
+      }
+      invoice_item = InvoiceItem.new(ii_details)
+
+      expect(invoice_item.id).to eq 6
+      expect(invoice_item.id).is_a? Integer
+    end
+
+    it 'returns the item id' do
+      ii_details = {
+        id: 6,
+        item_id: 7,
+        invoice_id: 8,
+        quantity: 1,
+        unit_price: BigDecimal(10.99, 4),
+        created_at: Time.now,
+        updated_at: Time.now
+      }
+      invoice_item = InvoiceItem.new(ii_details)
+
+      expect(invoice_item.item_id).to eq 7
+    end
+
+    it 'returns the invoice id' do
+      ii_details = {
+        id: 6,
+        item_id: 7,
+        invoice_id: 8,
+        quantity: 1,
+        unit_price: BigDecimal(10.99, 4),
+        created_at: Time.now,
+        updated_at: Time.now
+      }
+      invoice_item = InvoiceItem.new(ii_details)
+
+      expect(invoice_item.invoice_id).to eq 8
+    end
+
+    it 'returns the quantity' do
+      ii_details = {
+        id: 6,
+        item_id: 7,
+        invoice_id: 8,
+        quantity: 1,
+        unit_price: BigDecimal(10.99, 4),
+        created_at: Time.now,
+        updated_at: Time.now
+      }
+      invoice_item = InvoiceItem.new(ii_details)
+
+      expect(invoice_item.quantity).to eq 1
+    end
+
+    it 'returns the unit price' do
+      ii_details = {
+        id: 6,
+        item_id: 7,
+        invoice_id: 8,
+        quantity: 1,
+        unit_price: BigDecimal(10.99, 4),
+        created_at: Time.now,
+        updated_at: Time.now
+      }
+      invoice_item = InvoiceItem.new(ii_details)
+
+      expect(invoice_item.unit_price).to eq 10.99
+    end
+
+    it 'returns a Time instance for date invoice_item was created' do
+      ii_details = {
+        id: 6,
+        item_id: 7,
+        invoice_id: 8,
+        quantity: 1,
+        unit_price: BigDecimal(10.99, 4),
+        created_at: Time.now,
+        updated_at: Time.now
+      }
+      invoice_item = InvoiceItem.new(ii_details)
+
+      expect(invoice_item.created_at).is_a? Time
+    end
+
+    it 'returns a Time instance for date invoice_item updated' do
+      ii_details = {
+        id: 6,
+        item_id: 7,
+        invoice_id: 8,
+        quantity: 1,
+        unit_price: BigDecimal(10.99, 4),
+        created_at: Time.now,
+        updated_at: Time.now
```

Comment:
I would suggest using just '2011-01-01' or something since Time.now is not a correct format with the CSV files and I don't think testing these data points as instances of Time would be very helpful in the future. 

I'm approving the PR because it at least tests that the instance is populating times-stamps with something but maybe later we can revisit this and test it more accurately to what we would want to expect in a real situation.

Reply:
If you're ok with it for now, I'll reserve making that time test more specific, just based on the interaction pattern 
`created_at - returns a Time instance for the date the invoice item was first created` 
seems from that testing the class of time should suffice for now. Definitely agree that making it more specific in the future could be useful

Reply:
@NoahZinter I actually had to refactor our times in the Item class and ItemRepo to avoid the pitfall Richard mentioned here. Just FYI. I think the spec harness will cause you some issues here later.